frontend/src/pages/CourseSelector/CourseDescription/PrerequisiteTree/PrerequisiteTree.jsx

```diff
@@ -0,0 +1,266 @@
+          return {
+            id: edge.id,
+            label: 'unlocks',
+          };
+        default:
+          return {
+            id: edge.id,
+          };
+      }
+    });
+  };
+
+  const bringLabelsToFront = (graphInstance) => {
+    // bring edges with labels to front 
+    graphInstance.getEdges().filter(e => e._cfg.model.hasOwnProperty('label')).forEach(e => e.toFront());
+    // Repaint the graph after shifting
+    graphInstance.paint();
+  };
+
+  const generateTreeGraph = (graphData) => {
+    const container = ref.current;
+    const treeGraphInstance = new G6.TreeGraph({
+      container: container,
+      width: container.scrollWidth,
+      height: container.scrollHeight,
+      fitView: true,
+      layout: {
+        type: 'mindmap',
+        direction: 'H',
+        getVGap: () => 0,
+        getHGap: () => 100,
+        getSide: (node) => {
+          if (node.data.rootRelationship === relationship.PREREQ) return "left";
+          return "right";
+        },
+      },
+      defaultNode: {
+        type: "rect",
+        style: {
+          radius: 5,
+          fill: "#9254de",
+          stroke: "#9254de",
+          cursor: "pointer",
+        },
+        labelCfg: {
+          style: {
+            fill: "#fff",
+            fontFamily: "Arial",
+            fontSize: 14,
+            fontWeight: "bold",
+            cursor: "pointer",
+          },
+        },
+      },
+      defaultEdge: {
+        type: "cubic-horizontal", // polyline could be used but pivots are unreliable with lots of courses
+        color: "#A6A6A6",
+        size: 1,
+        // style: {
+        //   offset: 100,
+        // },
+        labelCfg: {
+          refX: 25,
+          position: 'start',
+          autoRotate: true,
+          style: {
+            fill: "#595959",
+            fontFamily: "Arial",
+            fontWeight: "bold",
+            fontSize: 14,
+            background: {
+              fill: '#ffffff',
+              padding: [2, 2, 2, 2],
+              radius: 5,
+            },
+          },
+        },
+      },
+    });
+
+    setGraph(treeGraphInstance);
+
+    treeGraphInstance.data(graphData);
+
+    updateEdges(treeGraphInstance, graphData);
+    
+    treeGraphInstance.render(graphData);
+    
+    bringLabelsToFront(treeGraphInstance);
+    
+    treeGraphInstance.on("node:click", (event) => {
+      // open new course tab
+      const node = event.item;
+      const { _cfg: { model: { label } } } = node;
+      dispatch(addTab(label));
+    });
+  };
+
+  const updateTreeGraph = (graphData) => {
+    graph.changeData(graphData);
+    bringLabelsToFront(graph);
+  };
+
+  /* MAIN */
+  const setupGraph = async (c) => {
+    setLoading(true);
+    let unlocks = [];
+    let prereqs = [];
+
+    // get all courses a course c unlocks
+    // must pass in no courses 
+    const { startYear } = planner;
+    const { programCode, specs } = degree;
+    const specialisations = {};
```

Comment:
I know this is a very hacky workaround coz of the BE but add a TODO comment for later :D. `TODO: Create stub payload when getting coursesUnlockedWhenTaken due to BE bug?`

Reply:
Wait could we use the courseChildren enpoint instead? https://circlesapi.csesoc.app/docs#/courses/course_children_courses_courseChildren__course__get

<img width="1454" alt="Screen Shot 2022-07-19 at 6 56 49 pm" src="https://user-images.githubusercontent.com/65492346/179710520-7e3abe1f-537c-4968-9958-5e8d3a005608.png">